proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -583,6 +587,24 @@ public void handle(SetCompression setCompression) throws Exception
         server.getCh().setCompressionThreshold( setCompression.getThreshold() );
     }
 
+    @Override
+    public void handle(ResourcePackRequest resourcePackRequest) throws Exception
+    {
+        ResourcePackRequestEvent requestEvent =
+            new ResourcePackRequestEvent( server, con, resourcePackRequest.getUrl(), resourcePackRequest.getHash(), resourcePackRequest.isForced(), resourcePackRequest.getPromptMessage() );
+
+        // If the event is cancelled or if two servers use the same resource pack, do not send the packet
+        if ( ( con.getResourcePackHash() != null && resourcePackRequest.getHash().equals( con.getRequestedResourcePackHash() ) ) || bungee.getPluginManager().callEvent( requestEvent ).isCancelled() )
+        {
+            // Send a success response to the server to make sure the client doesn't get disconnected if the resource pack
+            // is forced
+            server.unsafe().sendPacket( new ResourcePackResponse( resourcePackRequest.getHash(), ResourcePackResponseEvent.Response.SUCCESSFULLY_LOADED.ordinal() ) );
```

Comment:
I don't think this is right, I think it's clientside that checks the resource pack was received

Reply:
this won't call the ResourcePackResponseEvent because we're sending the packet to the server. ResourcePackResponseEvent is called in UpstreamBridge. either way yes, the kick is handled clientside and i can remove that code.